### ISSUE TITLE
bug: Side Menu expose-aside-when directive doesn't work

### ISSUE THREAD

**Issue by TheCodeDestroyer** [driftyco/ionic#3244]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>all</span>  
<span ionic-description>
**Version**: v1.0.0-rc.0-nightly-1120

Now there is an issue when my menu is closed and I press the button it opens the side menu and immediately closes it.
The menu works if I don't include this directive...

You can see this issue in [CodePen](http://codepen.io/TheCodeDestroyer/pen/VYGLKG)

The issue is in this function: `self.exposeAside()` as this if clause doesn't return:
````js
if (!(self.left && self.left.isEnabled) && !(self.right && self.right.isEnabled)) return;
````

I thank you in advance.</span>

<span is-issue-template></span>

**Comment by TheCodeDestroyer** [comment]
Did a bit more digging and it's not `self.exposeAside()` issue. I checked v1.0.0-beta.14-nightly-1000
and the difference between RC nightly is that `checkAsideExpose()` is being called all the time, which then closes the side menu since the if clause fails to be true.

**Comment by tolu360** [comment]
This bug is also in v1.0.0-rc.0, and in nightlies after the version 1.0.0-beta.14-nightly-1042.

**Comment by shprink** [comment]
same over here

**Comment by ekonstantinidis** [comment]
+1. Just updated from beta 14 to rc.0 and now the side menu is broken because of `expose-aside-when="large"`. If I remove it, then the side menu is working properly.

**Comment by vgrafe** [comment]
+1, would love to see this feature working as planned.

**Comment by sbilharz** [comment]
+1!

**Comment by mixersoft** [comment]
I saw this as well in rc0. I just added the following line for now

     self.exposeAside = function(shouldExposeAside) {
        if (!(self.left && self.left.isEnabled) && !(self.right && self.right.isEnabled)) return;
        # do not close when the side-menu was opened by toggle()
        if (angular.element($ionicBody.get()).hasClass('menu-open')) return

**Comment by Riron** [comment]
+1. [This line](https://github.com/driftyco/ionic/commit/27298e9209a718450c48048592dc5cf8ffb139f8#diff-7f8a45a06c53978f6e2571220899a0eaR157) seems to be the problem, as said in [this issue](https://github.com/driftyco/ionic/issues/3219)

**Comment by animungo** [comment]
+1

**Comment by dudeko** [comment]
@mixersoft's line fixes the problem I'm having.

My problem:
I've got expose-aside-when="large" and, on a small screen, if I have focus on an input with the virtual keyboard opened, and then I tap to toggle the left menu, the left menu autocloses just after blur happens on the input and the virtual keyboard closes.

What I gathered is that every time the keyboard appears/disappears, the size of the view changes, and an evaluation is made whether the side menu should be opened or closed because of the expose-aside-when directive.

You can replicate this problem by:
-creating a new sidemenu project, 
-putting an input on playlists.html, 
-putting expose-aside-when="large" on the ion-side-menu inside menu.html,
-running the app, activating the keyboard by tapping the input and then toggling the side menu. It should open and close automatically.

I can't seem to find any other good way of dealing with this. I've tried doing a timeout to delay the menu toggle, and it kinda works, but the time it takes for the keyboard to close varies and depends on how much content I have on my view. Also, this isn't the ideal solution, as it would feel sluggish for the user to tap and have to wait 2 seconds or so for the side menu to open.

**Comment by mlynch** [comment]
Duplicate of #2328 which is now fixed and will be in 1.2
